external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2470,6 +2645,302 @@ fn make_arg_types<S: std::fmt::Display, F: Fn() -> S>(
     given
 }
 
+fn check_call_target(
+    context: &mut Context,
+    call_loc: Loc,
+    is_macro_call: Option<Loc>,
+    declared_macro_modifier: Option<Loc>,
+    declared: Loc,
+    f: FunctionName,
+) {
+    let decl_is_macro = declared_macro_modifier.is_some();
+    if is_macro_call.is_some() == decl_is_macro {
+        return;
+    }
+
+    let macro_call_loc = is_macro_call.unwrap_or(call_loc);
+    let decl_loc = declared_macro_modifier.unwrap_or(declared);
+    let call_msg = if decl_is_macro {
+        format!(
+            "'{f}' is a macro function and must be called with a `!`. \
+            Try replacing with '{f}!'"
+        )
+    } else {
+        format!(
+            "'{f}' is not a macro function and cannot be called with a `!`. \
+            Try replacing with '{f}'"
+        )
+    };
+    let decl_msg = if decl_is_macro {
+        "Declared a macro function here"
+    } else {
+        "Declared a normal (non-macro) function here"
+    };
```

Comment:
```suggestion
    let decl_msg = if decl_is_macro {
        "Macro function is declared here"
    } else {
        "Normal (non-macro) function is declared here"
    };
```

Reply:
I like the new, suggested wording better :) But I do have this stilted "Declared _ here" grammar all over, we should maybe note to clean those up when we see 'em